### PR TITLE
[tmf] avoid binding TMF port for user UDP

### PR DESCRIPTION
**Pull request by wgtdkp**

### CODE REVIEW

File: src/core/net/udp6.cpp [hunk 1]
```diff
@@ -229,10 +229,11 @@ Error Udp::Bind(SocketHandle &aSocket, const SockAddr &aSockAddr, otNetifIdentif
         do
         {
             aSocket.mSockName.mPort = GetEphemeralPort();
+        } while (aSocket.mSockName.mPort == Tmf::kUdpPort
```
Comment (simonlingoogle) [inline src/core/net/udp6.cpp:232]: nit: Should we move this logic to `GetEphemeralPort` so that it never returns `Tmf::kUdpPort` ?

Comment (wgtdkp) [reply src/core/net/udp6.cpp:232]: SGTM!

**Review by bukepo — APPROVED** [review]
LGTM:+1:

**Review by simonlingoogle — APPROVED** [review]
lgtm 👍

**Review by jwhui — APPROVED** [review]

**Pull request merged**